### PR TITLE
sql/expression/function: implement char_length and length functions

### LINKED ISSUE #715: Add LENGTH function (for strings)
- Add LENGTH (and CHAR_LENGTH?) for strings
- LENGTH can also operate on binary data.
- Both exist in MySQL and Spark SQL

Current workaround: `ARRAY_LENGTH(SPLIT(repository_id, ''))`

### PR DESCRIPTION
**Pull request by erizocosmico**
Fixes #715

### CODE REVIEW

File: sql/expression/function/length_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+)
+
+func TestLength(t *testing.T) {
```
Comment (juanjux) [inline sql/expression/function/length_test.go]: Maybe also add a test with an empty string.

**Comment by erizocosmico** [comment]
All fixed

**Comment by juanjux** [comment]
LGTM, needs rebase.

**Comment by erizocosmico** [comment]
Rebased
